app/Models/BeatmapsetEvent.php

```diff
@@ -92,6 +92,7 @@ public static function search($rawParams = [])
 
         $query = static::limit($params['limit'])->offset($pagination['offset']);
         $searchByUser = present($rawParams['user'] ?? null);
+        $searchByBeatmapset = present($rawParams['beatmapset'] ?? null);
```

Comment:
also just something to note: when searching by user, certain event types related to moderation are filtered out (see later usage of `$searchByUser`). I'm not sure if the same should apply when searching by beatmapset. the behavior is kind of confusing to me considering everything is public when not directly searching by user...

Reply:
i assume this comment is resolved by https://github.com/ppy/osu-web/pull/10552/commits/d5dac9355196499b094966daf4baf65691d26ec4? not sure what the relevance is otherwise, i'm not doing any filtering out..

Reply:
yeah your new code isn't filtering anything out, I'm saying that's what `$searchByUser` currently does, and so maybe searching by beatmapset also *should*. someone else needs to give their input on it, I don't fully understand the purpose

Reply:
it's for some moderation events where the event itself is visible but user_id isn't public. Showing them when filtered by user id means leaking the user id itself.